src/pytorch_tabular/tabular_model_tuner.py

```diff
@@ -333,6 +342,20 @@ def tune(
                     else:
                         result = tabular_model_t.evaluate(validation, verbose=False)
                         params.update({k.replace("test_", ""): v for k, v in result[0].items()})
+
+                    if best_model is None:
```

Comment:
Same as the other model sweep PR: 
make `tabular_model.datamodule = None` before deep copying it, and as the end of the loop, set `best_model.datamodule` = tabular_model.datamodule`

Reply:
Changed.